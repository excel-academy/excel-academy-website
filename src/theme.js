import { themeColor } from '../site-config';

const purples = ['#7b79b0', '#6f6da5', '#eeeefc', '#1c2070'];
const greens = ['#d2eced', '#132c2d'];
const fontSizes = [12, 14, 16, 18, 20, 24, 30, 38, 62];
const space = [0, '2rem', '3rem', '4rem', '5rem', '7rem', '32rem', '40rem', '48rem', '56rem', '64rem'];
export const breakpoints = {
  sm: 0,
  mobile: '30em',
  tablet: '46em',
  desktop: '61.25em',
  wide: '80em',
  extrawide: '105em',
};

export default {
  space,
  fontSizes,
  breakpoints,
  maxWidths: ['48em', '60em', '80em'],
  colors: {
    theme: themeColor,
    purples,
    greens,
  },
  buttons: {
    primary: {
      color: '#fff',
      backgroundColor: themeColor,
      fontSize: fontSizes[4],
      borderColor: '#fff',
      paddingRight: space[4],
      paddingLeft: space[4],
      '&:hover': {
        backgroundColor: '#000',
      },
    },
    outline: {
      color: themeColor,
      backgroundColor: 'transparent',
      boxShadow: 'inset 0 0 2px',
    },
  },
};
