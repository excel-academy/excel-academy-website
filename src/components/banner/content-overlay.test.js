import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Heading } from 'rebass/styled-components';
import theme from '../../theme';
import ContentOverlayBanner from './content-overlay';
import image from '../../utils/mockBannerImg';

describe('ContentOverlayBanner', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <ContentOverlayBanner image={image.src} imagedesc={image.alt} contentAlign="center" orient="left" imagePosition="center" bg="blue" fullscreen>
          <Heading fontWeight={300} fontSize="5rem" lineHeight={1} mb={1}>Ernie Bello</Heading>
        </ContentOverlayBanner>
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
