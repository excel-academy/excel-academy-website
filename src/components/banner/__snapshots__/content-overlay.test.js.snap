// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentOverlayBanner renders correctly 1`] = `
<DocumentFragment>
  .c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 20px;
  font-family: heading;
  font-weight: heading;
  line-height: heading;
  font-weight: 300;
  font-size: 5rem;
  line-height: 1;
  margin-bottom: 2rem;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 2rem;
  z-index: 1;
}

.c6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c6 .gatsby-image-wrapper {
  height: 100%;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 2rem;
  background-color: blue;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-x: hidden;
  text-align: center;
  min-height: 100vh;
}

.c1 .c5 img {
  object-position: center !important;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .c2 {
  width: 40rem;
}

.c0 .c5 {
  opacity: 0.5;
}

@media screen and (min-width:30em) {
  .c1 {
    padding: 2rem;
  }
}

@media screen and (min-width:46em) {
  .c1 {
    padding: 3rem;
  }
}

@media screen and (min-width:61.25em) {
  .c1 {
    padding: 4rem;
  }
}

@media screen and (min-width:80em) {
  .c1 {
    padding: 5rem;
  }
}

@media screen and (min-width:105em) {
  .c1 {
    padding: 7rem;
  }
}

<section
    class="c0 c1"
    orient="left"
  >
    <div
      class="c2 c3"
    >
      <h2
        class="c4"
        font-size="5rem"
        font-weight="300"
      >
        Ernie Bello
      </h2>
    </div>
    <div
      class="c5 c6"
    >
      <div
        class=" gatsby-image-wrapper"
        style="position: relative; overflow: hidden;"
      >
        <div
          style="width: 100%; padding-bottom: 66.66666666666667%;"
        />
        <picture>
          <source
            sizes="(max-width: 600px) 100vw, 600px"
            srcset="some srcSet"
          />
          <img
            alt="banner"
            loading="lazy"
            sizes="(max-width: 600px) 100vw, 600px"
            src="test-file-stub"
            srcset="some srcSet"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;source srcset="some srcSet" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;img loading="lazy" sizes="(max-width: 600px) 100vw, 600px" srcset="some srcSet" src="test-file-stub" alt="banner" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
